engine/src/main/java/com/datatorrent/stram/plan/physical/PhysicalPlan.java

```diff
@@ -895,8 +895,10 @@ private void redoPartitions(PMapping currentMapping, String note)
         addedPartitions.add(newPartition);
       } else {
         // check whether mapping was changed
+        int currentPartitionsSize = mainPC.currentPartitions.size();
         for (DefaultPartition<Operator> pi : mainPC.currentPartitions) {
-          if (pi == newPartition && pi.isModified()) {
+          if (pi == newPartition && (pi.isModified() ||
```

Comment:
How that is different from a case when current partition size is 2 and new partition size is 3 or more and it includes 2 current partitions? Is the behavior different because there were no unifiers? But unifiers may never come into the picture or be introduced several operators downstream. The problem seems to be in the definePartitions() that returns the current partition. I may be wrong, but I don't see why it is necessary to handle current partition size equal to one differently in the modified code.